readthedocs/core/static-src/core/js/doc-embed/search.js

```diff
@@ -53,20 +64,92 @@ function attach_elastic_search_query(data) {
-                                if (index < 3) {
-                                    // Show up to 3 results for search
-                                    var content = highlight.content[index];
-                                    var content_text = xss(content);
-                                    var contents = $('<div class="context">');
-
-                                    contents.html("..." + content_text + "...");
-                                    contents.find('em').addClass('highlighted');
-                                    list_item.append(contents);
+                        for (var j = 0; j < inner_hits.length; j += 1) {
+
+                            var contents = $('<div class="context">');
+
+                            // if the result is page section
+                            if(inner_hits[j].type === "sections") {
+                                
+                                var section = inner_hits[j];
+                                var subtitle = $('<div class="rtd_search_subtitle">');
+                                var subtitle_link = $('<a href="' + link + "#" + section._source.id + '">');
+                                var section_content = $('<span>')
+                                
+                                if (section.highlight) {
+                                    if (section.highlight["sections.title"]) {
+                                        subtitle_link.html(
+                                            xss(section.highlight["sections.title"][0])
+                                        );
+                                    } else {
+                                        subtitle_link.html(
+                                            section._source.title
+                                        )
+                                    }
+                                
+                                    if (section.highlight["sections.content"]) {
+                                        section_content.html(
+                                            "... " +
+                                            xss(section.highlight["sections.content"][0]) +
+                                            " ..."
+                                        )
+                                    } else {
+                                        section_content.html(
+                                            section._source.content.substring(0, 100) +
+                                            " ..."
+                                        )
+                                    }
                                 }
+                                
+                                subtitle.html(subtitle_link);
+                                contents.append(subtitle);
+                                contents.append(section_content)
+                                contents.find('em').addClass('highlighted')
```

Comment:
Can this not use underscore templates, also?

Reply:
We can.
I have updated the PR.